bears/general/IndentationBear.py

```
@@ -0,0 +1,209 @@
+                                the range of the outermost indentation while
+                                last being the range of the most
+                                nested/innermost indentation.
+                                Equal level indents appear in the order of
+                                first encounter or left to right.
+                                OR a string stating un-matched indents.
+        """
+        ranges = []
+        for open_indent in indent_types:
+            close_indent = indent_types[open_indent]
+            open_pos = list(IndentationBear.get_valid_chars(
+                                           file, open_indent, annotation_dict))
+            close_pos = list(IndentationBear.get_valid_chars(
+                                           file, close_indent, annotation_dict))
+
+            to_match = len(open_pos) - 1
+            while to_match >= 0:
+                close_index = 0
+                while close_index < len(close_pos):
+                    if(open_pos[to_match].position
+                       <= close_pos[close_index].position):
+                        ranges.append(
+                            SourceRange.from_absolute_position(
+                                            filename,
+                                            open_pos[to_match],
+                                            close_pos[close_index]))
+                        close_pos.remove(close_pos[close_index])
+                        open_pos.remove(open_pos[to_match])
+                        to_match -= 1
+                        break
+                    close_index += 1
+                if((len(close_pos) == 0 and to_match != -1) or
+                   (len(close_pos) != 0 and to_match == -1)):
+                    # None to specify unmatched indents
+                    return "Unmatched " + open_indent + ", " +\
+                           close_indent + " pair"
```

it's very weird to retun a str in one situation and another thing in the other and then use it as an error message. You could just raise an exception here which would make more sense I think.